src/components/QueryBuilder.vue

```diff
@@ -82,31 +51,17 @@ export default Vue.extend( {
 			this.$store.dispatch( 'incrementMetric', metric );
 		},
 		validate(): void {
+			// TODO: Clean up FormValue <-> Query relation
```

Comment:
Maybe we can transform this into a getter `isQueryBuilderValid`? Then we can look at its result in this method and set the state accordingly. But that can be done in a follow-up

Reply:
Yes, This PR is already getting pretty big.